### PR TITLE
[WIP] Add package tracking history from postur.is

### PR DESCRIPTION
**Pull request by koddsson**
New endpoint to track package coming to Iceland, using postur.is as a datasource.
### Checklist
- [x] Write tests
- [ ] Write documentation
- [x] Handle 404
- [x] Handle incorrect formatted tracking number.
- [ ] Add package metadata

### CODE REVIEW

**Comment by koddsson** [comment]
I'm not gonna add the package metadata or write documentation in this PR.

**Comment by MiniGod** [comment]
LGTM

**Comment by koddsson** [comment]
Need to merge https://github.com/apis-is/apis/pull/376 before this codeship test will pass.

File: endpoints/tracking/tests/integration_test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import request from 'request'
+import assert from 'assert'
+
+describe('tracking', () => {
+  it('should return a 404 when the tracking number can\'t be found', (done) => {
+    request.get('http://localhost:3101/tracking/derp', (error, response) => {
```
Comment (koddsson) [inline endpoints/tracking/tests/integration_test.js]: TODO: Mock the test

File: endpoints/tracking/tests/integration_test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import request from 'request'
+import assert from 'assert'
+
+describe('tracking', () => {
+  it('should return a 404 when the tracking number can\'t be found', (done) => {
```
Comment (koddsson) [inline endpoints/tracking/tests/integration_test.js]: TODO: Add a test for when tracking actually works

File: endpoints/tracking/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-param-reassign */
+
+import request from 'request'
+import cheerio from 'cheerio'
+import app from '../../server'
+
+app.get('/tracking/:trackingNumber', (req, res) => {
```
Comment (MiniGod) [inline endpoints/tracking/index.js]: Is `/tracking` maybe too general?

File: endpoints/tracking/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      .reduce((sum, curr) => {
+        sum[curr.date] = curr.action
+        return sum
+      }, {})
+
+    if (Object.keys(history).length === 0) {
+      return res.status(404).json({})
+    }
+
+    return res.json({
```
Comment (MiniGod) [inline endpoints/tracking/index.js]: Cache it?

Comment (koddsson) [reply endpoints/tracking/index.js]: I'm wondering how long tho. Presumably people would want to know the status of their package right now? Maybe a cache of a minute would help if we are getting bombarded with requests tho?

**Comment by koddsson** [comment]
I had a bunch of issues running the tests for this. Waiting on #380 to land so I can run the test in isolation and have speedier debugging. There might be more PRs in the meantime to get this even better.

File: endpoints/tracking/index.js (hunk 4 above)
Comment (MiniGod) [reply endpoints/tracking/index.js]: I'd say any caching is good. 1 minute sounds good. We can always tweak it later if we feel the need.

**Comment by koddsson** [comment]
This and basically any new endpoint needs a better way on how to mock the data sources and not mock the `localhost`. I'm working on it now and hopefully it won't take a bunch of time.

**Comment by koddsson** [comment]
>This and basically any new endpoint needs a better way on how to mock the data sources and not mock the `localhost`. I'm working on it now and hopefully it won't take a bunch of time.

I've started this in #403.

**Comment by koddsson** [comment]
It seeeeems like postur.is is now a client side thing and thus you can just get the data from this handy URL `https://www.postur.is/Umbraco/Surface/PackageTrackerApi/Handle -d "number=TRACKING_NUMBER&lang=IS"`

That kind of makes this endpoint hard to develop and useless! 😸

**Pull request closed without merging**